test/e2e/specs/site-editor/style-variations.spec.js

```diff
@@ -0,0 +1,299 @@
+			postId: 'gutenberg-test-themes/style-variations//index',
+			postType: 'wp_template',
+		} );
+		await siteEditorStyleVariations.applyPinkVariation();
+		await editor.openPreviousGlobalStylesPanel();
+
+		await siteEditorStyleVariations.openColorsPanel();
+		expect(
+			await siteEditorStyleVariations.getBackgroundColorValue()
+		).toBe( 'rgb(202, 105, 211)' );
+		expect( await siteEditorStyleVariations.getTextColorValue() ).toBe(
+			'rgb(74, 7, 74)'
+		);
+
+		await editor.openPreviousGlobalStylesPanel();
+		await siteEditorStyleVariations.openTypographyPanel();
+		await siteEditorStyleVariations.openTextPanel();
+
+		expect( await siteEditorStyleVariations.getFontSizeHint() ).toBe(
+			'Medium(px)'
+		);
+	} );
+
+	test( 'should apply custom colors and font sizes in a variation', async ( {
+		admin,
+		editor,
+		siteEditorStyleVariations,
+	} ) => {
+		await admin.visitSiteEditor( {
+			postId: 'gutenberg-test-themes/style-variations//index',
+			postType: 'wp_template',
+		} );
+		await siteEditorStyleVariations.applyYellowVariation();
+		await editor.openPreviousGlobalStylesPanel();
+		await siteEditorStyleVariations.openColorsPanel();
+
+		expect(
+			await siteEditorStyleVariations.getBackgroundColorValue()
+		).toBe( 'rgb(255, 239, 11)' );
+		expect( await siteEditorStyleVariations.getTextColorValue() ).toBe(
+			'rgb(25, 25, 17)'
+		);
+
+		await editor.openPreviousGlobalStylesPanel();
+		await siteEditorStyleVariations.openTypographyPanel();
+		await siteEditorStyleVariations.openTextPanel();
+
+		expect( await siteEditorStyleVariations.getFontSizeHint() ).toBe(
+			'(Custom)'
+		);
+		expect( await siteEditorStyleVariations.getCustomFontSizeValue() ).toBe(
+			'15'
+		);
+	} );
+
+	test( 'should apply a color palette in a variation', async ( {
+		admin,
+		editor,
+		siteEditorStyleVariations,
+	} ) => {
+		await admin.visitSiteEditor( {
+			postId: 'gutenberg-test-themes/style-variations//index',
+			postType: 'wp_template',
+		} );
+		await siteEditorStyleVariations.applyPinkVariation();
+		await editor.openPreviousGlobalStylesPanel();
+		await siteEditorStyleVariations.openColorsPanel();
+		await siteEditorStyleVariations.openPalettePanel();
+
+		expect(
+			await siteEditorStyleVariations.getThemePalette()
+		).toMatchObject( [
+			{
+				color: 'rgb(74, 7, 74)',
+				name: 'Foreground',
+			},
+			{
+				color: 'rgb(202, 105, 211)',
+				name: 'Background',
+			},
+			{
+				color: 'rgba(204, 0, 255, 0.77)',
+				name: 'Awesome pink',
+			},
+		] );
+	} );
+
+	test( 'should reflect style variations in the styles applied to the editor', async ( {
+		admin,
+		page,
+		siteEditorStyleVariations,
+	} ) => {
+		await admin.visitSiteEditor( {
+			postId: 'gutenberg-test-themes/style-variations//index',
+			postType: 'wp_template',
+		} );
+		await siteEditorStyleVariations.applyYellowVariation();
+		const frame = await page.frameLocator(
+			'css=.edit-site-visual-editor iframe'
+		);
+		const paragraph = await frame.locator(
```

Comment:
Neither `.locator` nor `.frameLocator` returns a promise. It's a common mistake when first migrating to Playwright 😆 .